splice_spark/src/test/scala/com/splicemachine/spark/splicemachine/NativeTransformationsTest.scala

```diff
@@ -89,7 +89,7 @@ class NativeTransformationsTest extends FunSuite with TestContext with BeforeAnd
     assert("SortMergeJoin".r.findAllMatchIn(plan).length == 2)
   }
 
-  ignore("complex project on join over join") {
+  test("complex project on join over join") {
     val df = splicemachineContext.df(
       """select s1.colsequence * 10 from sys.systables s1
```

Comment:
> Based on the overflow limitations I thought this shouldn't work, and it turns out we are promoting the Int to BigInteger. Where does that happen? That might be a good strategy to deal with integer overflow if we then convert it back to Integer after making sure it didn't overflow, but we can't return a BigInteger were we are expecting an integer, right?

@dgomezferro The result data type of 's1.colsequence * 10' is promoted to BigInt in NumericTypeCompiler.resolveArithmeticOperation:

		if (java.sql.Types.TINYINT == highTypeId ||
	            java.sql.Types.SMALLINT == highTypeId ||
		    java.sql.Types.INTEGER == highTypeId)
		{
			return new DataTypeDescriptor(
					TypeId.getBuiltInTypeId(java.sql.Types.BIGINT),
					nullable
				);
		}

This type of operation should never overflow because max integer * max integer is still less than max bigint.  For other operations that could overflow, I am not sure how to accomplish  "making sure it didn't overflow" programatically.  We can revisit this later, but I agree we need some solution to enable more cases.

Reply:
Oh I see, in that case the overflow logic is a quite a bit less restrictive than I thought.